cmake/Codegen.cmake

```diff
@@ -122,6 +122,22 @@ if (INTERN_BUILD_ATEN_OPS)
   list(LENGTH CPU_CAPABILITY_NAMES NUM_CPU_CAPABILITY_NAMES)
   math(EXPR NUM_CPU_CAPABILITY_NAMES "${NUM_CPU_CAPABILITY_NAMES}-1")
 
+  IF(CXX_AVX512_FOUND)
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DHAVE_AVX512_CPU_DEFINITION")
+
+    check_cxx_compiler_flag("-mno-avx256-split-unaligned-load -mno-avx256-split-unaligned-store" COMPILER_SUPPORTS_NO_AVX256_SPLIT)
+    IF(COMPILER_SUPPORTS_NO_AVX256_SPLIT)
+      SET(CPU_NO_AVX256_SPLIT_FLAGS "-mno-avx256-split-unaligned-load -mno-avx256-split-unaligned-store")
+    ENDIF(COMPILER_SUPPORTS_NO_AVX256_SPLIT)
```

Comment:
Is this section needed? Wouldn't it be covered by the code at line 109 above?

Reply:
But that's only set for AVX2 case. no?

Reply:
Is there a reason we wanna disable split? It looks like it is beneficial on some CPU models but not on some other CPU models. https://gcc.gnu.org/bugzilla/show_bug.cgi?id=49089

Reply:
Check the comments in original file?

Reply:
Which comment?

Reply:
https://stackoverflow.com/questions/52626726/why-doesnt-gcc-resolve-mm256-loadu-pd-as-single-vmovupd#tab-top

Reply:
OK, thanks. I interpret that as we wanna disable split because we mostly run on aligned memory.

Reply:
@llyfacebook are you aware of AVX512 machines that *dont* support AVX2?

Reply:
No, AVX512 machine should support AVX2, AVX. and AVX2 machine supports AVX. 

I think I agree that I need to get rid of the CPU_NO_AVX256_SPLIT_FLAGS definition here, since it is already defined and we always build for all CPU_CAPABILITIES.